tests/test_reordering.f90

```diff
@@ -0,0 +1,180 @@
+program test_reorder
```

Comment:
Having a high level test for the reorders is great, I think we can remove the low level tests for reorders on CUDA once we're sure this works for CUDA too. I think it is not yet ready to run the CUDA kernels though, mainly because of the `data_d` component in `cuda_field_t`. I see that you don't have the hardware to test so I'll be happy to make those minor changes later on and get this working with CUDA kernels as well.

Reply:
ah yes, sure. Feel free to edit this PR to make it work there

Reply:
I think it is all sorted now. For now I haven't removed the low level CUDA tests in cuda folder because they include performance tests as well. Once we have a mechanism for checking performance in the high level code low level reorder tests in cuda folder can be removed completely. Checking the performance is very helpful for me to maintain a good performance, so I prefer not losing this ability.

Reply:
The only thing is in the test program there is a mix of 3/4 spacing, it would be nice to have it consistent with the main codebase. Apart from this I think the PR is in good shape, please merge after taking a look at the changes I pushed.

Reply:
Yes, sure. Once we have a dedicated place to put performance tests we can clean up the duplicate unit tests.